op-batcher/batcher/driver.go

```diff
@@ -483,7 +489,13 @@ func (l *BatchSubmitter) clearState(ctx context.Context) {
 }
 
 // publishTxToL1 submits a single state tx to the L1
-func (l *BatchSubmitter) publishTxToL1(ctx context.Context, queue *txmgr.Queue[txRef], receiptsCh chan txmgr.TxReceipt[txRef]) error {
+func (l *BatchSubmitter) publishTxToL1(ctx context.Context, queue *txmgr.Queue[txRef], receiptsCh chan txmgr.TxReceipt[txRef], daWaitGroup *sync.WaitGroup, daSemaphore *semaphore.Weighted) error {
+	// Acquire semaphore or return immediately if it is at capacity
+	// we don't want to pull data out of the state if we won't be able to send it
+	if l.Config.UseAltDA && !daSemaphore.TryAcquire(1) {
```

Comment:
should we overload error, or just create a new error. Since there is only one caller function, I will prefer new error